src/jaxsim/rbda/contacts/visco_elastic.py

```diff
@@ -0,0 +1,860 @@
+
+        Returns:
+            A `ViscoElasticContactsParams` instance with the specified parameters.
+
+        Note:
+            The `damping_ratio` parameter allows to operate on the following conditions:
+            - ξ > 1.0: over-damped
+            - ξ = 1.0: critically damped
+            - ξ < 1.0: under-damped
+        """
+
+        soft_contacts_params = SoftContactsParams.build_default_from_jaxsim_model(
+            model=model,
+            standard_gravity=standard_gravity,
+            static_friction_coefficient=static_friction_coefficient,
+            max_penetration=max_penetration,
+            number_of_active_collidable_points_steady_state=number_of_active_collidable_points_steady_state,
+            damping_ratio=damping_ratio,
+        )
+
+        return ViscoElasticContactsParams.build(
+            K=soft_contacts_params.K,
+            D=soft_contacts_params.D,
+            static_friction=soft_contacts_params.mu,
+            max_squarings=max_squaring,
+            soft_contacts_non_linear_p=soft_contacts_non_linear_p,
+            soft_contacts_non_linear_q=soft_contacts_non_linear_q,
+        )
+
+    def valid(self) -> jtp.BoolLike:
+        """
+        Check if the parameters are valid.
+
+        Returns:
+            `True` if the parameters are valid, `False` otherwise.
+        """
+
+        return (
+            jnp.all(self.K >= 0.0)
+            and jnp.all(self.D >= 0.0)
+            and jnp.all(self.static_friction >= 0.0)
+            and jnp.all(self.max_squarings > 0)
+            and jnp.all(self.soft_contacts_non_linear_p >= 0.0)
+            and jnp.all(self.soft_contacts_non_linear_q >= 0.0)
+        )
+
+    def __hash__(self) -> int:
+
+        from jaxsim.utils.wrappers import HashedNumpyArray
+
+        return hash(
+            (
+                HashedNumpyArray.hash_of_array(self.K),
+                HashedNumpyArray.hash_of_array(self.D),
+                HashedNumpyArray.hash_of_array(self.static_friction),
+                hash(self.max_squarings),
+                hash(self.soft_contacts_non_linear_p),
+                hash(self.soft_contacts_non_linear_q),
+            )
+        )
+
+    def __eq__(self, other: ViscoElasticContactsParams) -> bool:
+
+        if not isinstance(other, ViscoElasticContactsParams):
+            return False
+
+        return hash(self) == hash(other)
+
+
+@jax_dataclasses.pytree_dataclass
+class ViscoElasticContacts(ContactModel):
+    """Visco-elastic contacts model."""
+
+    parameters: ViscoElasticContactsParams = dataclasses.field(
+        default_factory=ViscoElasticContactsParams
+    )
+
+    terrain: jax_dataclasses.Static[Terrain] = dataclasses.field(
+        default_factory=FlatTerrain
+    )
+
+    @jax.jit
+    def compute_contact_forces(
+        self,
+        model: js.model.JaxSimModel,
+        data: js.data.JaxSimModelData,
+        *,
+        dt: jtp.FloatLike,
+        joint_forces: jtp.VectorLike | None = None,
```

Comment:
```suggestion
        joint_force_references: jtp.VectorLike | None = None,
```
Hi @diegoferigo, one early comment: in the other contact models we called this parameter `joint_force_references`, maybe we should maintained aligned the naming conventions (either align also the other models or align this to the others)

Reply:
Ow you're right, I forgot. Thanks for noticing. I need to align also #247 then.